sources/engine/Stride.Graphics/FastTextRenderer.cs

```diff
@@ -209,14 +217,13 @@ public unsafe void End([NotNull] GraphicsContext graphicsContext)
 
                 //Draw the strings
                 var constantInfos = new RectangleF(GlyphWidth, GlyphHeight, DebugSpriteWidth, DebugSpriteHeight);
+                Span<VertexPositionNormalTexture> vertexPositionSpan = stackalloc VertexPositionNormalTexture[4* textInfo.Text.Length];
```

Comment:
Logic expects `mappedVertexBufferPointer` to receive the data, you should create a span from that, so something like `new Span<VertexPositionNormalTexture>(mappedVertexBufferPointer, amountOfItemsTheBufferContains);`

Also, `stackalloc` should be used with care, I used it in the benchmark as it wasn't that important there, but if you don't know in advance how much bytes you're going to allocate (the string in this case might be *very* long), you may be filling the thread's stack resulting in a `StackOverflowException` see [stackalloc](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/stackalloc)

Reply:
Btw, Thanks everyone for your help 😀. @Eideren Thanks, I've updated my branch now. When it comes to amountOfItemsTheBufferContains do you had in mind something like in line 214? (Unsafe.InitBlock..)

Reply:
Almost, that's the amount of bytes, `new Span<T>(void*, int)`'s second argument is the amount of `T` or items in the collection, see [here](https://learn.microsoft.com/en-us/dotnet/api/system.span-1.-ctor?view=net-7.0#system-span-1-ctor(system-void*-system-int32)). So it should just be `new((void*)mappedVertexBufferPointer, VertexBufferLength)`

Reply:
Oh my 🤦 You're right. Updated now 😀